### PR TITLE
Add translation of DateTime.AddMilliseconds

### LINKED ISSUE #1496: DateTime.AddMilliseconds method not supported
### The issue

For some reason I need to use the `DateTime.AddMilliseconds` method to remove the millisecond in the `DateTime`, and this cause an exception.

```
Translation of method 'System.DateTime.AddMilliseconds' failed.
```

And I'd like to know if this method will be translated in the future.

### Further technical details
 
Pomelo.EntityFrameworkCore.MySql version:5.0.1

### PR DESCRIPTION
**Pull request by iamSmallY**
Close #1496 

I use 
```mysql
date_add(@dt, interval num * 1000 microsecond);
```
to translate the
```C#
dt.AddMilliseconds(num);
```

Since the `AddMilliseconds` method will round the parameter to the nearest integer, I also add four tests for round up and round down.

### CODE REVIEW

File: src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs [hunk 1]
```diff
@@ -59,7 +63,11 @@ public MySqlDateTimeMethodTranslator(ISqlExpressionFactory sqlExpressionFactory)
                                 new SqlExpression[]
                                 {
                                     _sqlExpressionFactory.Fragment("INTERVAL"),
-                                    _sqlExpressionFactory.Convert(arguments[0], typeof(int)),
+                                    datePart.Equals("microsecond")
+                                        ? _sqlExpressionFactory.Multiply(
+                                            new SqlConstantExpression(Expression.Constant(1000), new IntTypeMapping("int")),
```
Comment (lauxjpn) [inline src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs:68]: Please use `_sqlExpressionFactory.Constant(1000)` here.

**Review by lauxjpn — CHANGES_REQUESTED** [review]
Thanks for adding the translation!
The SQL Server provider implements it as well.

File: src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs [hunk 2]
```diff
@@ -59,7 +63,11 @@ public MySqlDateTimeMethodTranslator(ISqlExpressionFactory sqlExpressionFactory)
                                 new SqlExpression[]
                                 {
                                     _sqlExpressionFactory.Fragment("INTERVAL"),
-                                    _sqlExpressionFactory.Convert(arguments[0], typeof(int)),
+                                    datePart.Equals("microsecond")
+                                        ? _sqlExpressionFactory.Multiply(
+                                            _sqlExpressionFactory.Constant(1000, new IntTypeMapping("int32", DbType.Int32)),
```
Comment (lauxjpn) [inline src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs:68]: Is the explicit type mapping really necessary? I would have assumed, that it works just fine without it.

**Review by lauxjpn — CHANGES_REQUESTED** [review]
Also please rebase your changes on top of `master`, instead of `5.0-maint`. Thanks!

File: src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs [hunk 3]
```diff
@@ -3,11 +3,13 @@
 
 using System;
 using System.Collections.Generic;
+using System.Data;
 using System.Reflection;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Diagnostics;
 using Microsoft.EntityFrameworkCore.Query;
 using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
+using Microsoft.EntityFrameworkCore.Storage;
```
Comment (lauxjpn) [inline src/EFCore.MySql/Query/Internal/MySqlDateTimeMethodTranslator.cs:12]: The two additional `using` statements should not be needed in the current code.

**Review by lauxjpn — CHANGES_REQUESTED** [review]

**Review by lauxjpn — APPROVED** [review]

**Comment by lauxjpn** [comment]
@iamSmallY Thank you for your contribution!

**Comment by iamSmallY** [comment]
No thanks😀

**Pull request merged**